functions_lab.js

```diff
@@ -0,0 +1,418 @@
+//
+console.log('=========answer#8=========')
+
+function multiples (num) {
+  let num2 = []
+for (let i = 1; i <= num; i++) {
+  if (i % 3 === 0 && i % 5 === 0) {
+    num2.push("FizzBuzz")
+} else if (i % 3 === 0) {
+  num2.push("Fizz")
+} else if (i % 5 === 0) {
+    num2.push("Buzz")
+} else {
+    num2.push(i)
+}
+
+  }
+  return num2
+}
+
+console.log(multiples(90))
+//
+// ## Question 9
+//
+// Write a function named `reverse` that takes an array as a parameter.
+//The function should return an array in reverse order.
+//
+//
+// Example:
+//
+// Input: `reverse([1, 2, 3])`
+//
+// Output: `[3, 2, 1]`
+
+console.log('=========answer#9===========')
+let arr = [1, 2, 3];
+
+const reverse = (arr) => {
+let reverseArr = [];
+for (let num of arr) {
+reverseArr.unshift(num)
+
+}
+return reverseArr
+}
+console.log(reverse(arr))
+
+
+
+
+//
+//
+// ## Question 10
+//
+// Write a function that prints out the most frequently occurring element in an
+//array.
+//
+// Example:
+//
+// Input: `[2,1,4,2,5,2,3,23,3,3,2,2,2,24,56,2,2,5,46,2]`
+//
+// Output: `2`
+ console.log("========answer#10=========")
+
+ let arr = [2,1,4,2,5,2,3,23,3,3,2,2,2,24,56,2,2,5,46,2]
+function mostFrequentElement (arr) {
+  return
+}
+
+  let myArr = {
+}
+
+
+ for (let i = 0; i < arr.length; i++) {
+  let currNum = arr[i]
+  if ( currNum !== ",") {
+  }
+   if (myArr[currNum]=== undefined) {
+     myArr[currNum] = 1;
+   } else {
+     myArr[currNum]++
+   }
+
+   let mostNum;
+   let mostNumCount = 0
+
+   for (let num in myArr) {
+     if (myArr[num] > mostNumCount) {
+       mostNum = num
+       mostNumCount = myArr[num]
+
+     }
+
+return mostNum
+
+   // console.log(mostNum)
+   // console.log(mostNumCount)
+ }
+   console.log(mostFrequentElement([arr]))
+
+   // DOESN'T RUN
+
+
+
+
+
+//
+//
+// ## Question 11
+//
+// Write a function that sums all the even numbers in an array.
+//
+// Example:
+//
+// Input: `[1,2,5,64,23,22,91,3]`
+//
+// Output: `88`
+
+// console.log("========answer#11===========")
+//
+ let arr = [1,2,5,64,23,22,91,3];
+
+ const evenSum = (arr) => {
+
+       let sum = 0
+for (let i = 0; i < arr.length; i++) {
+  if (arr[i] % 2 === 0){
+        sum += arr[i]
+  }
+}
+    return sum
+ }
+ console.log(evenSum(arr))
+
+
+
+//
+//
+// ## Question 12
+//
+// Write a function that flips an object.  All of the keys are now values and
+//all of the values are now keys.
+//
+// Example:
+//
+// Input: `{1: "uno", 5: "cinco"}`
+//
+// Output: `{"uno": 1, "cinco": 5}
+
+// console.log('========answer#12==========')
+
+let obj = {1: "uno", 5: "cinco"}
+const flipObj = (obj) => {
+  let flip = {}
+  for (let key in obj) {
+    flip[obj[key]] = key;
+  }
+  return flip
+
+}
+console.log(flipObj(obj))
+//
+//
+// ## Question 13
+//
+// Write a function that determines if a value is inside of array.
+//
+// Example:
+//
+// Input: `[1,4,6,9,10], 5`
+//
+// Output: `false`
+
+console.log("========answer#13========")
+
+let theArr = [1, 4, 6, 9, 10,]
```

Comment:
This would stop the function in first iteration of the loop as 1, which is the first item inside array, would evaluate as true and return false. Once value is returned, both loop and function stop running.